Storage/Storage/SQL/BrowserDB.swift

```diff
@@ -8,11 +8,22 @@ public enum QuerySort {
     case None, LastVisit, Frecency
 }
 
+public class DateRange {
```

Comment:
Unfortunately, @wesj's code uses `as? DateRange`, which is not valid with a struct.  I don't intend to address this (really, your substantive comment about `FilterType`) as part of this pull request.